navigator/frontend/src/ui-core/src/ParameterForm/index.tsx

```diff
@@ -581,6 +589,77 @@ const ListInput = (props: ListInputProps): JSX.Element => {
   } else {
     return (<TypeErrorElement parameter={parameter} argument={argument} />);
   }
+};
+
+const MapInput = (props: MapInputProps): JSX.Element => {
+  const { argument, parameter, level, onChange, disabled, contractIdProvider, typeProvider } = props;
+  if (matchPrimitiveType(argument, parameter, 'map')) {
+    const elements = argument && argument.type === 'map' ? argument.value : [];
+    const elementType = parameter.args[0] || DamlLfTypeF.unit();
+    return (
+      <NestedForm level={level}>
+        {elements.map((entry, i) => (
+          <LabeledElement label={`entry[${i}]`} key={`entry[${i}]`}>
+            <NestedForm level={level + 1}>
+              <LabeledElement label={`key`} key={`key[${i}]`}>
+                <StyledTextInput
+                  type="string"
+                  disabled={disabled}
+                  placeholder="String"
+                  step="any"
+                  value={elements[i].key}
+                  onChange={(key) => {
+                    const newElements = elements.slice(0);
+                    newElements[i] = mapEntry((key.target as HTMLInputElement).value, newElements[i].value);
+                    onChange(DamlLfValueF.map(newElements));
+                  }}
+                />
+              </LabeledElement>
+              <LabeledElement label={`value`} key={`value[${i}]`}>
+                <ParameterInput
+                  contractIdProvider={contractIdProvider}
+                  typeProvider={typeProvider}
+                  parameter={elementType}
+                  name={`value[${i}]`}
```

Comment:
`name` is the full path of the property being edited. E.g., `foo.bar[0]` refers to first element of list given by the field `bar` in the record given by the field `foo` of the contract argument. It's not printed anywhere on the screen currently, but is useful in debugging. 

```suggestion
                  name={`${name}[${i}].value`}
```

`name` needs to be read from props: `const { ..., name } = props;`